lib/active_merchant/billing/gateways/beanstream.rb

```diff
@@ -64,148 +65,34 @@ module Billing #:nodoc:
-        add_recurring_service(post, options)
-        recurring_commit(post)
-      end
-
-      def cancel_recurring(options = {})
-        ActiveMerchant.deprecated RECURRING_DEPRECATION_MESSAGE
-
-        post = {}
-        add_recurring_operation_type(post, :cancel)
-        add_recurring_service(post, options)
-        recurring_commit(post)
-      end
-
-      def interac
-        @interac ||= BeanstreamInteracGateway.new(@options)
-      end
-
-      # To match the other stored-value gateways, like TrustCommerce,
-      # store and unstore need to be defined
-      def store(payment_method, options = {})
-        post = {}
-        add_address(post, options)
-
-        if payment_method.respond_to?(:number)
-          add_credit_card(post, payment_method)
-        else
-          post[:singleUseToken] = payment_method
-        end
-        add_secure_profile_variables(post, options)
-
-        commit(post, true)
-      end
-
-      #can't actually delete a secure profile with the supplicated API. This function sets the status of the profile to closed (C).
-      #Closed profiles will have to removed manually.
-      def delete(vault_id)
-        update(vault_id, false, {:status => "C"})
-      end
+    #
+    # Modification on 2018-03-12:
+    # New feature: integrated the ipp gateway into beanstream gateway, use this class as an unified entrance for both the gateways
+    # Get the :region field from options parameter, :region: 0 - US & CA, 1 - AU
+    # Proess payment request from US & CA using beanstream service (default)
+    # Proess payment request from AU using ipp service
+    # Will remove ipp gateway file
 
-      alias_method :unstore, :delete
+    class BeanstreamGateway < Gateway
+      attr_accessor :region
+      include Empty
 
-      # Update the values (such as CC expiration) stored at
-      # the gateway.  The CC number must be supplied in the
-      # CreditCard object.
-      def update(vault_id, payment_method, options = {})
-        post = {}
-        add_address(post, options)
-        if payment_method.respond_to?(:number)
-          add_credit_card(post, payment_method)
-        else
-          post[:singleUseToken] = payment_method
-        end
-        options.merge!({:vault_id => vault_id, :operation => secure_profile_action(:modify)})
-        add_secure_profile_variables(post,options)
-        commit(post, true)
-      end
+      def self.new(options = {})
+        # region: 0 - US & CA, 1 - AU & NZ
+        # I want to add this logic to judge the region from shipping & billing address
+        # but don't know if it is a correct way.
+        # I think this can be decided by shopify which will integrate with.
 
-      def supports_scrubbing?
-        true
-      end
+        # [ options[:billing_address], options[:shipping_address] ].compact.each do |address|
+        #   next if empty?(address[:country])
+        #   unless ['AU', 'NZ'].include?(address[:country])
+        #     @region = 1
```

Comment:
when is IPP required? Does it have to do with the shipping and/or billing address?

Reply:
I am not clear about tht. The staff from ipp just asked for the unify gateway for different regions' orders. So, I think that if there is a way to judge the region automatically? Could you give me some suggestion about it?